src/openads-appnexus/service/ReplayEventBus.js

```diff
@@ -0,0 +1,80 @@
+import {observerErrorThrown} from './observerErrorThrown'
+
+class ReplayEventBus {
+  constructor() {
+    this._observers = new Map()
+    this._pendingEvents = new Map()
+  }
+
+  register({eventName, observer}) {
+    if (!eventName) {
+      throw new Error('Event Name is required')
+    }
+    if (typeof observer !== 'function') {
+      throw new Error('Observer must be a function')
+    }
+    if (!this._observers.has(eventName)) {
+      this._observers.set(eventName, [observer])
+      this._replayPendingEvent({eventName})
```

Comment:
As we already talked in person, here you should only do something like this:
```javascript
this._pendingEvents.forEach(event => observer({event})
```
So any new Observer registered, must be notified with pending events in memory, always.
If Refresh use case, clear pending events in the right moment, forEach here will not do anything and the find method will wait to a new event raised.

Reply:
updated to process all raised events for a new registered observer